### PR TITLE
Switch to on-demand instances for AWS tests on scheduled runs.

### PR DESCRIPTION
**Pull request by marseel**
More context in the issue.

Related: https://github.com/cilium/cilium/issues/29365

### CODE REVIEW

**Comment by marseel** [comment]
run with spot set to false: https://github.com/cilium/cilium/actions/runs/6981426957/job/18998539323

File: .github/workflows/conformance-aws-cni.yaml [hunk 1]
```diff
@@ -199,6 +199,7 @@ jobs:
           region: ${{ matrix.region }}
           owner: "${{ steps.vars.outputs.owner }}"
           version: ${{ matrix.version }}
+          spot: ${{ github.event_name == 'schedule' }}
```
Comment (nbusseneau) [inline .github/workflows/conformance-aws-cni.yaml]: Isn't this doing the opposite of what we want? 🤔

Comment (marseel) [reply .github/workflows/conformance-aws-cni.yaml]: omg yes :facepalm:
fixed, thanks.

**Review by nbusseneau — APPROVED** [review]

**Review by viktor-kurchenko — APPROVED** [review]

**Comment by marseel** [comment]
Both relevant tests passed, marking as ready-to-merge

**Pull request merged**